tools/karl.cpp

```diff
@@ -1038,6 +1068,114 @@ void handle_arguments (int argc, char ** argv)
+      obPath.copy(argv_config_files[flag_cnt], obPath.size(), 0);
+      argv_config_files[flag_cnt][obPath.size()] = 0; // null string terminator 
+      flag_cnt++; 
+    } 
+
+    madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                          "Found karl config file: %s\n", 
+                          full_path);                   
+       
+    // read each line of the text formatted file in, each line contains 
+    // one flag followed by a space then the param if there is a parameter
+    while (std::getline(file, flag_param))
+    {
+      flag = flag_param.substr(0, flag_param.find_first_of(" "));
+      if (flag_param.find_first_of(" ") == std::string::npos)
+      {
+        param = "";
+      }
+      else
+      {
+        param = flag_param.substr(flag_param.find_first_of(" ") + 1,
+        flag_param.length() - (flag_param.find_first_of(" ") + 1)); 
+      }      
+      madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                            "Flag: %s, Param: %s\n", 
+                            flag, param);         
+      
+      if ((flag == "-cf" || flag == "--config-file") && file_cnt >= 0)
+      {
+        config_file = param;        
+        
+        madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                              "Found user karl config file flag, param: %s, File Cnt: %s\n", 
+                              config_file, file_cnt);                  
+        file_cnt--;
+ 
+        return (load_config_file(argv_config_files, config_file, file_cnt, flag_cnt));        
+      }    
+      else 
+      {
+        if (file_cnt >= 0)
+        {
+          
+          madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                                "Found a flag saving now...\n");                 
+          argv_config_files.push_back(nullptr);
+          argv_config_files[flag_cnt] = new char[flag.size() + 1];
+          flag.copy(argv_config_files[flag_cnt], flag.size(), 0);
+          argv_config_files[flag_cnt][flag.size()] = 0; // null string terminator
+          flag_cnt++;
+
+          if (param != "")
+          {
+            madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                                "Found a param saving now...\n");             
+            argv_config_files.push_back(nullptr);
+            argv_config_files[flag_cnt] = new char[param.size() + 1];
+            param.copy(argv_config_files[flag_cnt], param.size(), 0); 
+            argv_config_files[flag_cnt][param.size()] = 0; // null string terminator         
+            flag_cnt++;
+          }
+
+        }      
+      }
+    }
+    file.close();
+    handle_arguments(argv_config_files.size(), &argv_config_files[0]);
+  
+    // garbage cleanup
+    for (unsigned int arg_index = 0; arg_index < argv_config_files.size(); arg_index++)
```

Comment:
Use range-based for (https://en.cppreference.com/w/cpp/language/range-for) whenever appropriate, like here.

Reply:
This chunk of code is gone, thanks tho.